src/test/java/com/google/cloud/resourcemanager/ResourceManagerImplTest.java

```diff
@@ -387,73 +438,210 @@ public void testTestPermissions() {
+    }
+  }
+
+  @Test
+  public void testClearOrgPolicy() {
+    String exceptionMessage = "Should fail because the organization policy doesn't exist.";
+    Mockito.doThrow(new ResourceManagerException(404, exceptionMessage))
+        .when(resourceManagerMock)
+        .clearOrgPolicy(RESOURCE, ORG_POLICY_INFO);
+    try {
+      resourceManagerMock.clearOrgPolicy(RESOURCE, ORG_POLICY_INFO);
+      fail();
+    } catch (RuntimeException expected) {
+      assertEquals(exceptionMessage, expected.getMessage());
+    }
+  }
+
+  @Test
+  public void testGetEffectiveOrgPolicy() {
+    Mockito.when(resourceManagerMock.getEffectiveOrgPolicy(RESOURCE, CONSTRAINTS))
+        .thenReturn(ORG_POLICY_INFO);
+    OrgPolicyInfo policyInfo = resourceManagerMock.getEffectiveOrgPolicy(RESOURCE, CONSTRAINTS);
+    assertEquals(CONSTRAINTS, policyInfo.getConstraint());
+    assertEquals(BOOLEAN_POLICY, policyInfo.getBoolPolicy());
+    assertEquals(ETAG, policyInfo.getEtag());
+    assertEquals(LIST_POLICY, policyInfo.getPolicies());
+    assertEquals(UPDATE_TIME, policyInfo.getUpdateTime());
+    assertEquals(VERSION, policyInfo.getVersion());
+  }
+
+  @Test
+  public void testGetEffectiveOrgPolicyWithException() {
+    String exceptionMessage = "Not Found";
+    Mockito.doThrow(new ResourceManagerException(404, exceptionMessage))
+        .when(resourceManagerMock)
+        .getEffectiveOrgPolicy(RESOURCE, CONSTRAINTS);
+    try {
+      resourceManagerMock.getEffectiveOrgPolicy(RESOURCE, CONSTRAINTS);
+      fail();
+    } catch (RuntimeException expected) {
+      assertEquals(exceptionMessage, expected.getMessage());
+    }
+  }
+
+  @Test
+  public void testGetOrgPolicy() {
+    Mockito.when(resourceManagerMock.getOrgPolicy(RESOURCE, CONSTRAINTS))
+        .thenReturn(ORG_POLICY_INFO);
+    OrgPolicyInfo policyInfo = resourceManagerMock.getOrgPolicy(RESOURCE, CONSTRAINTS);
+    assertEquals(CONSTRAINTS, policyInfo.getConstraint());
+    assertEquals(BOOLEAN_POLICY, policyInfo.getBoolPolicy());
+    assertEquals(LIST_POLICY, policyInfo.getPolicies());
+    assertEquals(UPDATE_TIME, policyInfo.getUpdateTime());
+    assertEquals(VERSION, policyInfo.getVersion());
+  }
+
+  @Test
+  public void testGetOrgPolicyWithException() {
+    String exceptionMessage = "Not Found";
+    Mockito.doThrow(new ResourceManagerException(404, exceptionMessage))
+        .when(resourceManagerMock)
+        .getOrgPolicy(RESOURCE, CONSTRAINTS);
+    try {
+      resourceManagerMock.getOrgPolicy(RESOURCE, CONSTRAINTS);
+      fail();
+    } catch (RuntimeException expected) {
+      assertEquals(exceptionMessage, expected.getMessage());
+    }
+  }
+
+  @Test
+  public void testListAvailableOrgPolicyConstraints() throws IOException {
+    ListResult<Constraint> expectedResult =
+        ListResult.of(CURSOR, ImmutableList.of(CONSTRAINT_INFO.toProtobuf()));
+    Mockito.when(resourceManagerRpcMock.listAvailableOrgPolicyConstraints(NAME, EMPTY_RPC_OPTIONS))
```

Comment:
You've mocked out the method you're trying to test.

Reply:
done